shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Metadata.hs

```diff
@@ -0,0 +1,143 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Ledger.ShelleyMA.Metadata
+  ( Metadata (..),
+    pattern Metadata,
+  )
+where
+
+import Cardano.Binary
+  ( FromCBOR (..),
+    ToCBOR (..),
+  )
+import Cardano.Crypto.Hash (hashWithSerialiser)
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Crypto (Crypto)
+import Cardano.Ledger.Era (Era)
+import Cardano.Ledger.ShelleyMA (MaryOrAllegra, ShelleyMAEra)
+import Cardano.Ledger.ShelleyMA.Scripts ()
+import Control.DeepSeq (deepseq)
+import Data.Coders
+import Data.Map.Strict (Map)
+import Data.MemoBytes
+import Data.Sequence.Strict (StrictSeq)
+import Data.Typeable (Typeable)
+import Data.Word (Word64)
+import GHC.Generics (Generic)
+import NoThunks.Class
+import Shelley.Spec.Ledger.MetaData
+  ( MetaDataHash (..),
+    MetaDatum,
+    ValidateMetadata (..),
+    validMetaDatum,
+  )
+
+-- | Raw, un-memoised metadata type
+data MetadataRaw era = MetadataRaw
+  { mdScriptPreimages :: !(StrictSeq (Core.Script era)),
+    -- | Unstructured metadata "blob"
```

Comment:
so sorry to bike-shed a word in a comment, and a variable name.  I myself had been using the word "unstructured" for this component, but @dcoutts let me know that actually we've been elsewhere been calling this "structured metadata" and this will confuse folks. `mdScriptPreimages` is great, though. :)